.github/workflows/frontend-docs-generation.yml

```diff
@@ -0,0 +1,31 @@
+name: Generate Documentation for the frontend
+
+on:
+    push:
+        branches:
+            - '**'
+    pull_request:
+        branches:
+            - '**'
+
+jobs:
+    generate-docs:
+        runs-on: ubuntu-latest
+        steps:
+            - name: Clone project
+              uses: actions/checkout@v3
+
+            - name: Install typedoc
+              run: yarn add typedoc
```

Comment:
This step is not required, because typedoc is already present in the package.json dev dep. change this with a yarn install for ci